### ISSUE TITLE
Documentation editor adds tests to yml

### ISSUE THREAD

**Issue by smitsrr** [AltimateAI/vscode-dbt-power-user#1268]
### Expected behavior

We converted our ymls to use the new `data_tests:` syntax that is expected in dbt 1.8, however, when running and saving documentation via the extension it re-adds the tests using the old syntax, which I must then remove. 

### Actual behavior

What is added to the documentation `.yml` file:
``` 
columns:
      - name: sfdc_user_id
        description: The unique identifier of the SFDC user
        data_tests:
          - not_null:
              severity: error
          - unique:
              severity: error
        data_type: varchar
        tests:
          - not_null
          - unique
```

### Steps To Reproduce

go to model
click `documentation editor`
click 'sync with database'
click 'save'

### Operating System

mac

### dbt version

1.8.3

### dbt Adapter

snowflake

### dbt Power User version

Published 2020-08-18, 23:09:28 Last released 2024-06-26, 20:37:24 Last updated 2024-07-03, 11:25:38

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

**Comment by anandgupta42** [comment]
We will pick it up soon @smitsrr .

**Comment by anandgupta42** [comment]
@smitsrr, a fix was released in version 0.42.0. Could you please confirm?
